src/test/kotlin/com/igorwojda/integer/countupanddown/challenge.kt

```diff
@@ -4,7 +4,7 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun countUpAndDown(n: Int): List<Int> {
-    TODO("not implemented")
+    return List(2*n+1){i -> if(i<=n) i else 2*n-i}
```

Comment:
Simplify this. Use ranges, and collectors/copying (https://kotlinlang.org/docs/reference/collection-aggregate.html), and plus operator (https://kotlinlang.org/docs/reference/collection-plus-minus.html). Can actually be an easy to read one-liner.